projects/apache-commons-configuration/YAMLConfigurationReadFuzzer.java

```diff
@@ -1,22 +1,65 @@
 import com.code_intelligence.jazzer.api.FuzzedDataProvider;
 
-import java.io.InputStream;
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 
 import org.apache.commons.configuration2.YAMLConfiguration;
 import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.yaml.snakeyaml.LoaderOptions;
 
 public class YAMLConfigurationReadFuzzer {
-    public static void fuzzerTestOneInput(byte[] data) {
+    public static void fuzzerTestOneInput(FuzzedDataProvider data) {
+        // Create helper objects from fuzzer data
+        final boolean useInputStream = data.consumeBoolean();
+        final boolean useLoaderOptions = data.consumeBoolean();
+
         // Create needed objects
-        YAMLConfiguration yamlConfig = new YAMLConfiguration();
-        InputStream inputStream = new ByteArrayInputStream(data);
+        final YAMLConfiguration yamlConfig = new YAMLConfiguration();
+        yamlConfig.setLogger(null); // disable logging
 
+        // the actual fuzzing starts here
         try {
-            yamlConfig.read(inputStream);
+            final InputStream inputStream;
+            final InputStreamReader reader;
+            if (useInputStream) {
+                if (useLoaderOptions) {
+                    final LoaderOptions loaderOptions = createLoaderOptions(data);
+                    inputStream = new ByteArrayInputStream(data.consumeBytes(Integer.MAX_VALUE));
+                    yamlConfig.read(inputStream, loaderOptions);
+                } else {
+                    inputStream = new ByteArrayInputStream(data.consumeBytes(Integer.MAX_VALUE));
+                    yamlConfig.read(inputStream);
+                }
+            } else {
+                if (useLoaderOptions) {
+                    final LoaderOptions loaderOptions = createLoaderOptions(data);
+                    inputStream = new ByteArrayInputStream(data.consumeBytes(Integer.MAX_VALUE));
+                    reader = new InputStreamReader(inputStream);
+                    yamlConfig.read(reader, loaderOptions);
```

Comment:
I don't think all this `InputStreamReader` part is needed as `read()` will take any InputStream: https://github.com/apache/commons-configuration/blob/master/src/main/java/org/apache/commons/configuration2/YAMLConfiguration.java#L110

Reply:
I don't think I understand the problem.
Yes, `read()` can take any `InputStream`, but there is also an overloaded variant takes takes a `Reader` and I wanted to increase the coverage of my fuzz target.
Should I refrain from doing that?

Reply:
Yes, but it all use the same code. `read(final Reader in)` is the same as `read(final InputStream in)`. Snakeyaml is handling the different types here.

Reply:
Okay, noted.